discord/commands/context.py

```diff
@@ -83,85 +78,74 @@ class ApplicationContext(discord.abc.Messageable):
     command: :class:`.ApplicationCommand`
         The command that this context belongs to.
     """
+    command: Optional[ApplicationCommand]
+
+    def __init__(
+        self,
+        bot: Bot,
+        interaction: Interaction,
+        *,
+        command: Optional[ApplicationCommand] = None,
+        args: List[Any] = None,
+        kwargs: Dict[str, Any] = None,
+        **kwargs2
```

Comment:
I'm confused on why there are two `kwargs` variables?

Reply:
`kwargs` holds the arguments for the callback, while kwargs2 is supposed to be a catch-all for the extra context parameters

Reply:
Can't you just use `functools.partial`?

Reply:
I don't get how that might be used. Do you have an example?

Reply:
https://docs.python.org/3.8/library/functools.html#functools.partial
It's very straightforward to use. I can give my own example if this is not enough.

Reply:
I mean in the context of the parameters of a class. I don't understand how it can be used then